frappe/core/doctype/doctype/doctype.py

```diff
@@ -212,6 +213,14 @@ def set_default_translatable(self):
 			if d.translatable and not supports_translation(d.fieldtype):
 				d.translatable = 0
 
+	def enable_indexing_for_dashboard_links(self):
+		"""Enable indexing for outgoing links used in dashboard"""
+		for d in self.fields:
+			if d.fieldtype == "Link" and frappe.db.exists(
+				"DocType Link", {"parent": d.options, "link_doctype": self.name, "link_fieldname": d.fieldname}
+			):
+				d.search_index = 1
```

Comment:
We should check one more thing here, if this column is already part of multi-column index then we shouldn't duplicate index. 

I'll push changes

Reply:
For mariadb we need to check that there isn't ANY index on table where the field has `seq_in_index=1`